pkg/util/image_prepper.go

```diff
@@ -36,12 +38,42 @@ type Prepper interface {
 	SupportsImage() bool
 }
 
-func (p ImagePrepper) GetImage() (Image, error) {
+func getImage(prepper Prepper) (Image, error) {
+	imgPath, err := prepper.GetFileSystem()
+	if err != nil {
+		return Image{}, fmt.Errorf("error msg: %s", err.Error())
+	}
+
+	config, err := prepper.GetConfig()
+	if err != nil {
+		return Image{}, fmt.Errorf("error msg: %s", err.Error())
+	}
+
+	glog.Infof("Finished prepping image %s", prepper.GetSource())
+	return Image{
+		Source: prepper.GetSource(),
+		FSPath: imgPath,
+		Config: config,
+	}, nil
+}
+
+func (p *ImagePrepper) GetImage() (Image, error) {
 	glog.Infof("Starting prep for image %s", p.Source)
 	img := p.Source
 
 	var prepper Prepper
 
+	// first, respect prefixes on image names
+	if strings.HasPrefix(p.Source, "daemon://") {
```

Comment:
It seems like this should be redundant with the loop below.

Reply:
this was stripping off the prefix of the image source before creating the prepper, but since the rest of the logic was essentially redundant I moved this into the `SupportsImage()` method.